docs/topics/http/sessions.txt

```diff
@@ -275,26 +275,29 @@ You can edit it multiple times.
 
     .. method:: get_expiry_age()
 
-      Returns the number of seconds until this session expires. For sessions
-      with no custom expiration (or those set to expire at browser close), this
-      will equal :setting:`SESSION_COOKIE_AGE`.
+      Used by the session backends to determine the session expiry age in
+      seconds when saving the session.
 
-      This function accepts two optional keyword arguments:
+      An explicit ``expiry`` can be passed as an optional keyword argument,
+      to be used rather a previous set (or default) expiry.
 
-      - ``modification``: last modification of the session, as a
-        :class:`~datetime.datetime` object. Defaults to the current time.
-      - ``expiry``: expiry information for the session, as a
-        :class:`~datetime.datetime` object, an :class:`int` (in seconds), or
-        ``None``. Defaults to the value stored in the session by
-        :meth:`set_expiry`, if there is one, or ``None``.
+      If provided the correct ``modification`` (a :class:`~datetime.datetime`
+      object), ``get_expiry_age`` can be used to calculate the remaining
+      session time **only** when the session expiry has been set to an explicit
+      date, in all other cases returns the maximum session lifetime.
```

Comment:
Going over this again and again, I still lean towards just steering folks away with this kind of thing: 

```diff
diff --git a/docs/topics/http/sessions.txt b/docs/topics/http/sessions.txt
index c36554f22c..40b29d4e4a 100644
--- a/docs/topics/http/sessions.txt
+++ b/docs/topics/http/sessions.txt
@@ -288,13 +288,32 @@ You can edit it multiple times.
         ``None``. Defaults to the value stored in the session by
         :meth:`set_expiry`, if there is one, or ``None``.
 
+      .. note::
+
+        This method is used by session backends to determine the session expiry
+        age in seconds when saving the session. It is not really intended for
+        usage outside of that context.
+
+        In particular, whilst it is **possible** to determine the remaining
+        lifetime of a session **just when** you have the correct
+        ``modification`` value **and** the ``expiry`` is set as a ``datetime``
+        object, where you do have the ``modification``, it is more
+        straight-forward to calculate the expiry by-hand::
+
+            expires_at = modification + timedelta(seconds=settings.SESSION_COOKIE_AGE)
+
+        In all cases where ``expiry`` is something other than a ``datetime``
+        object, ``get_expiry_age()`` returns the maximum session lifetime,
+        which is likely not what you intend.
+
     .. method:: get_expiry_date()
 
       Returns the date this session will expire. For sessions with no custom
       expiration (or those set to expire at browser close), this will equal the
       date :setting:`SESSION_COOKIE_AGE` seconds from now.
 
-      This function accepts the same keyword arguments as :meth:`get_expiry_age`.
+      This function accepts the same keyword arguments as
+      :meth:`get_expiry_age`, and similar notes on usage apply.
 
     .. method:: get_expire_at_browser_close()
```

diff from current `main`. 

Anyone implementing a session backend can go and look at the source. For everyone else it feels like too much info, when what we really want to say is _Don't use this_. 

What do you think?

Reply:
Yeah I think that directs the users better, rather the warning should help direct them away.

With each of the original descriptions I added "on session save" in the first sentences. Which I hope is not more confusing. It's all better clarified in the note.
I'm becoming doc blind :joy: but I think we're getting there.